arch/xtensa/core/gen_zsr.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 Intel corporation
+# SPDX-License-Identifier: Apache-2.0
+import sys
+import re
+
+# Scratch register allocator.  Zephyr uses multiple Xtensa SRs as
+# scratch space for various special purposes.  Unfortunately the
+# configurable architecture means that not all registers will be the
+# same on every device.  This script parses a pre-cooked ("gcc -E
+# -dM") core-isa.h file for the current architecture and assigns
+# registers to usages.
+
+NEEDED = ("ALLOCA", "CPU", "FLUSH")
+
+coreisa = sys.argv[1]
+outfile = sys.argv[2]
+
+syms = {}
+
+def get(s):
+    return syms[s] if s in syms else 0
+
+with open(coreisa) as infile:
+    for line in infile.readlines():
+        m = re.match(r"^#define ([^ ]+) ?(.*)", line.rstrip())
```

Comment:
I think this fails to match multiple spaces or tabs:
```
#define       XCHAL...
```
Not a problem thanks to some `-dM` guarantee(s)?

Reply:
Yeah, the syntax is strict: exactly one space between "#define", the macro name, and its expansion.  It's not like it's specified anywhere in a document though.  Can't hurt to robustify.